### PR TITLE
Bugfix release v2.0.1

### PR DESCRIPTION
**Pull request by fsimonis**
## How to work with this template

* [x] assign a release manager, who takes care of the process
* [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.

## Step by step guide
* [x] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Add things, if necessary
   * Extract summary
   * Fix wording
   * Sort the entries lexicographically
* [x] Check code base w.r.t code formatting (run [`precice/tools/formatting/check-format`](https://github.com/precice/precice/blob/develop/tools/formatting/check-format)) and reformat if required (run [`precice/tools/formatting/format-all`](https://github.com/precice/precice/blob/develop/tools/formatting/format-all))
* [x] Create branch `bugfix-N` from develop. If needed, `git rebase develop`.
* [x] Open PR from `bugfix-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
* [ ] Fix potential problems in bugfix-branch (all)
* [x] Run `tools/releasing/bumpversion.sh MAJOR.MINOR.PATCH` to bump the version
* [x] Verify the version changes in:
   * [x] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
   * [x] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
   * [x] [debian changelog](https://github.com/precice/precice/blob/develop/tools/releasing/packaging/debian/changelog)
   * [ ] (do independent release, if necessary!) [Python bindings](https://github.com/precice/python-bindings)
   * [ ] (do independent release, if necessary!) [MATLAB bindings](https://github.com/precice/matlab-bindings)
* [ ] Draft message to mailing list
* [ ] Update documentation (all)
  * [ ] Update markdown configuration reference in wiki
* [ ] Approve the PR with at least two reviews (all)
* [ ] Merge PR to master 
* [ ] Tag release on master and verify by running `git describe --tags` (on GitHub, make sure to select the release branch as target)
* [ ] Merge back to develop and verify by running `git describe --tags`

## Regression Tests

Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289

* [ ] SU2 / CalculiX [flap_perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/SU2-CalculiX)
* [ ] OpenFOAM / OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
* [ ] OpenFOAM / CalculiX FSI [flap perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/OpenFOAM-CalculiX)
* [ ] OpenFOAM / CalculiX FSI - NP mapping in CalculiX [3D_Tube](https://github.com/precice/tutorials/tree/develop/FSI/3D_Tube/OpenFOAM-CalculiX)
* [ ] OpenFOAM / CalculiX / OpenFOAM CHT [heat_exchanger](https://github.com/precice/tutorials/tree/develop/CHT/heat_exchanger/buoyantSimpleFoam-CalculiX)
* [ ] OpenFOAM / deal.II [flap_perp_2D](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp_2D/OpenFOAM-deal.II)
* [ ] OpenFOAM / FEniCS [flap_perp](https://github.com/precice/tutorials/tree/master/FSI/flap_perp/OpenFOAM-FEniCS)
* [ ] OpenFOAM / FEniCS [cylinderFlap, only run first few minutes](https://github.com/precice/tutorials/tree/develop/FSI/cylinderFlap/OpenFOAM-FEniCS)
* [ ] OpenFOAM / FEniCS [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-fenics)
* [ ] OpenFOAM / Nutils [flow-over-plate](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils)
* [ ] FEniCS / FEniCS [partitioned-heat](https://github.com/precice/tutorials/tree/master/HT/partitioned-heat/fenics-fenics)
* [ ] MATLAB / MATLAB [ODEs](https://github.com/precice/matlab-bindings/tree/develop/tutorial)
* [ ] ExaFSA: Ateles / FASTEST
* [x] Alya
* [ ] 1D-ElasticTube [C++](https://github.com/precice/elastictube1d/tree/develop/cxx)
* [ ] 1D-ElasticTube [Python](https://github.com/precice/elastictube1d/tree/develop/python)
* [ ] SuperMUC
* [x] Solverdummy [C++](https://github.com/precice/precice/tree/develop/tools/solverdummies/cpp)
* [ ] Solverdummy [C](https://github.com/precice/precice/tree/develop/tools/solverdummies/c)
* [ ] Solverdummy [Fortran](https://github.com/precice/precice/tree/develop/tools/solverdummies/fortran)
* [ ] Solverdummy [Fortran 2003](https://github.com/precice/precice/tree/develop/tools/solverdummies/f2003)
* [ ] Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy)
* [ ] Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy)

## Post-release
* [ ] Generate packages
   * [ ] Latest Ubuntu LTS
   * [ ] Latest Ubuntu
   * [ ] Arch Linux AUR Package
* [ ] Update Spack recipe

### Release new version for bindings (to ensure compatibility with newest preCICE version)

- [ ] fortan-module?
- [ ] [MATLAB bindings](https://github.com/precice/matlab-bindings/compare/master...develop)
- [ ] [python bindings](https://github.com/precice/python-bindings/compare/master...develop)

### (only if breaking changes) Open PRs or issues `develop -> master` for all adapters

- [ ] athlet-adapter? 
- [ ] [calculix-adapter](https://github.com/precice/calculix-adapter/compare/master...develop)
- [ ] code_aster-adapter?
- [ ] [comsol-adapter](https://github.com/precice/comsol-adapter/compare/master...develop)
- [ ] [dealii-adapter](https://github.com/precice/dealii-adapter/compare/master...develop)
- [ ] lsdyna-adapter?
- [ ] [fenics-adapter](https://github.com/precice/fenics-adapter/compare/master...develop)
- [ ] [fluent-adapter](https://github.com/precice/fluent-adapter/compare/master...develop)
- [ ] mbdyn-adapter?
- [ ] [openfoam-adapter](https://github.com/precice/openfoam-adapter/compare/master...develop)
- [ ] [su2-adapter](https://github.com/precice/su2-adapter/compare/master...develop)

### (only if breaking changes) Open PRs or issues `develop -> master` for all other tools

- [ ] [aste](https://github.com/precice/aste/compare/master...develop)
- [ ] [elastictube1d](https://github.com/precice/elastictube1d/compare/master...develop)
- [ ] [tutorials](https://github.com/precice/tutorials/compare/master...develop)
- [ ] [systemtests](https://github.com/precice/systemtests/compare/master...develop)

### Marketing

* [ ] Send email and do marketing
* [ ] Tweet

### Misc

* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)

### CODE REVIEW

**Comment by uekerman** [comment]
Alya Flap testcase works. I tested NN and PETSc RBF mapping.

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 1]
```diff
@@ -214,7 +214,8 @@ PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::PetRadialBasisFctMapping(
   if (getDimensions() == 2) {
     _deadAxis = {xDead, yDead};
     PRECICE_CHECK(not(xDead and yDead), "You cannot choose all axes to be dead for a RBF mapping");
-    PRECICE_CHECK(not zDead, "You cannot dead out the z-axis if dimension is set to 2");
+    if (zDead)
+      PRECICE_WARN("Setting the z-axis to dead on a 2 dimensional problem has not effect.");
```
Comment (MakisH) [inline src/mapping/PetRadialBasisFctMapping.hpp]:
```suggestion
      PRECICE_WARN("Setting the z-axis to dead on a 2 dimensional problem has no effect.");
```

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 2]
```diff
@@ -672,7 +673,8 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(int inputDataID, int
         auto sigma = petsc::Vector::allocate(_matrixQ, "sigma", petsc::Vector::LEFT);
         if (not _QRsolver.solveTranspose(tau, sigma)) {
           KSPView(_QRsolver, PETSC_VIEWER_STDOUT_WORLD);
-          PRECICE_ERROR("RBF Polynomial linear system has not converged.");
+          PRECICE_ERROR("RBF Polynomial linear system has not converged. "
+                        "Try to fix axis-aligned mapping setups by marking perpendicular axis as dead.");
```
Comment (MakisH) [inline src/mapping/PetRadialBasisFctMapping.hpp]:
```suggestion
                        "Try to fix axis-aligned mapping setups by marking perpendicular axes as dead.");
```

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 3]
```diff
@@ -681,7 +683,8 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(int inputDataID, int
         utils::Event eSolve("map.pet.solveConservative.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
         if (not _solver.solve(au, out)) {
           KSPView(_solver, PETSC_VIEWER_STDOUT_WORLD);
-          PRECICE_ERROR("RBF linear system has not converged.");
+          PRECICE_ERROR("RBF linear system has not converged. "
+                        "Try to fix axis-aligned mapping setups by marking perpendicular axis as dead.");
```
Comment (MakisH) [inline src/mapping/PetRadialBasisFctMapping.hpp]:
```suggestion
                        "Try to fix axis-aligned mapping setups by marking perpendicular axes as dead.");
```

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 4]
```diff
@@ -733,7 +736,8 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(int inputDataID, int
       if (_polynomial == Polynomial::SEPARATE) {
         if (not _QRsolver.solve(in, a)) {
           KSPView(_QRsolver, PETSC_VIEWER_STDOUT_WORLD);
-          PRECICE_ERROR("Polynomial QR linear system has not converged.");
+          PRECICE_ERROR("Polynomial QR linear system has not converged. "
+                        "Try to fix axis-aligned mapping setups by marking perpendicular axis as dead.");
```
Comment (MakisH) [inline src/mapping/PetRadialBasisFctMapping.hpp]:
```suggestion
                        "Try to fix axis-aligned mapping setups by marking perpendicular axes as dead.");
```

File: src/mapping/RadialBasisFctMapping.hpp [hunk 5]
```diff
@@ -85,7 +85,7 @@ class RadialBasisFctMapping : public Mapping {
       _deadAxis[1] = yDead;
       PRECICE_CHECK(not(xDead && yDead), "You cannot choose all axis to be dead for a RBF mapping");
       if (zDead)
-        PRECICE_WARN("Setting the z-axis to dead on a 2 dimensional problem has not effect and will be ignored.");
+        PRECICE_WARN("Setting the z-axis to dead on a 2 dimensional problem has not effect.");
```
Comment (MakisH) [inline src/mapping/RadialBasisFctMapping.hpp]:
```suggestion
        PRECICE_WARN("Setting the z-axis to dead on a 2 dimensional problem has no effect.");
```

File: src/mapping/RadialBasisFctMapping.hpp [hunk 6]
```diff
@@ -194,8 +194,10 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
   }
 
   _qr = matrixCLU.colPivHouseholderQr();
-  if (not _qr.isInvertible())
-    PRECICE_ERROR("Interpolation matrix C is not invertible.");
+  if (not _qr.isInvertible()) {
+    PRECICE_ERROR("RBF interpolation matrix is not invertible! "
+                  "Try to fix axis-aligned mapping setups by marking perpendicular axis as dead.");
```
Comment (MakisH) [inline src/mapping/RadialBasisFctMapping.hpp]:
```suggestion
                  "Try to fix axis-aligned mapping setups by marking perpendicular axes as dead.");
```

**Comment by fsimonis** [comment]
@MakisH I fixed the typos and created a commit authored by you.

**Pull request merged**